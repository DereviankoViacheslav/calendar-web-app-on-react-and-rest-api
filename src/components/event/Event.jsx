import React from 'react';
import './Event.scss';
import moment from 'moment';
import PropTypes from 'prop-types';

function getStyle(event, heigthHour) {
  const heightOneMinute = heigthHour / 60;
  const startDay = moment(event.startDate).format('YYYY-MM-DD');
  const startEvent = moment(event.startDate);
  const endEvent = moment(event.endDate);
  const diffMinutes = endEvent.diff(startEvent, 'minutes');
  const top = startEvent.diff(startDay, 'minutes');

  return {
    backgroundColor: `${event.color}`,
    top: `${top * heightOneMinute}px`,
    height: `${diffMinutes * heightOneMinute}px`
  };
}

function Event({ event, onShowPopup, heigthHour }) {
  return (
    <div onClick={(e) => onShowPopup(e, event)}
      className="day-event"
      style={getStyle(event, heigthHour)}
    >
      <div className="day-event__title">{event.name}</div>
      <span>{`${moment(event.startDate).format('HH:mm')} - ${moment(event.endDate).format('HH:mm')}`}</span>
    </div>
  );
}

Event.propTypes = {
  event: PropTypes.object.isRequired,
  onShowPopup: PropTypes.func.isRequired,
  heigthHour: PropTypes.number.isRequired,
}

Event.defaultProps = {
  title: 'Default title',
}

export default Event;