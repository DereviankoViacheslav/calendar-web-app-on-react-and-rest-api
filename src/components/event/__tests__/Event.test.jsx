import React from 'react';
import { shallow } from 'enzyme';
import Event from '../Event'

describe('<Event />', () => {
  const props = {
    event: {
      id: '11',
      name: 'Event 1111',
      createDate: '2020-02-03T08:41:25+02:00',
      startDate: '2020-02-13T11:45:00+02:00',
      endDate: '2020-02-13T15:45:00+02:00',
      description: '',
      color: '#ff6c00',
    },
    onShowPopup: jest.fn(),
    heigthHour: 42,
  };

  it('Get snapshot', () => {
    const wrappedComponent = shallow(<Event {...props} />);
    expect(wrappedComponent).toMatchSnapshot();
  });

  it('should render day-event', () => {
    const wrappedComponent = shallow(<Event {...props} />);
    expect(wrappedComponent.find('.day-event').exists()).toBeTruthy();
  });
});