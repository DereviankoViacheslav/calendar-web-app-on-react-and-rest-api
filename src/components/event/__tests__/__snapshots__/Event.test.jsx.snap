// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Event /> Get snapshot 1`] = `
<div
  className="day-event"
  onClick={[Function]}
  style={
    Object {
      "backgroundColor": "#ff6c00",
      "height": "168px",
      "top": "493.49999999999994px",
    }
  }
>
  <div
    className="day-event__title"
  >
    Event 1111
  </div>
  <span>
    11:45 - 15:45
  </span>
</div>
`;
